pkg/deviceplugins/usb_device_plugin_helper.go

```diff
@@ -0,0 +1,207 @@
+		u.BCD = int(val)
+	case "DEVNAME":
+		u.DevicePath = filepath.Join("/dev", value)
+	default:
+		logrus.Printf("Skipping unknown key=value %s=%s\n", key, value)
+	}
+
+	return true
+}
+
+func WalkUSBDevices() (map[int][]*USBDevice, error) {
+	usbDevices := make(map[int][]*USBDevice, 0)
+	err := filepath.Walk("/sys/bus/usb/devices", func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// Ignore named usb controllers
+		if strings.HasPrefix(info.Name(), "usb") {
+			return nil
+		}
+		// We are interested in actual USB devices information that
+		// contains idVendor and idProduct. We can skip all others.
+		if _, err := os.Stat(filepath.Join(path, "idVendor")); err != nil {
+			return nil
+		}
+
+		// Get device information
+		if device := parseSysUeventFile(path); device != nil {
+			usbDevices[device.Vendor] = append(usbDevices[device.Vendor], device)
+		}
+		return nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return usbDevices, nil
+}
+
+func parseUSBSymLinkToPCIAddress(link string) string {
+	paths := strings.Split(link, "/")
+
+	if len(paths) < 3 {
+		return ""
+	}
+
+	pciAddress := paths[len(paths)-3 : len(paths)-2][0]
+	return pciAddress
+}
+
+func discoverPluggedUSBDevices() *LocalDevices {
+	usbDevices := make(map[int][]*USBDevice, 0)
+	err := filepath.Walk(pathToUSBDevices, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		// Ignore named usb controllers
+		if strings.HasPrefix(info.Name(), "usb") {
+			return nil
+		}
+		// We are interested in actual USB devices information that
+		// contains idVendor and idProduct. We can skip all others.
+		if _, err := os.Stat(filepath.Join(path, "idVendor")); err != nil {
+			return nil
+		}
+
+		// Get device information
+		if device := parseSysUeventFile(path); device != nil {
+			usbDevices[device.Vendor] = append(usbDevices[device.Vendor], device)
+		}
+		return nil
+	})
+
+	if err != nil {
+		logrus.Error("Failed when walking usb devices tree")
+	}
+	return &LocalDevices{devices: usbDevices}
+}
+
+func parseSelector(s *v1.USBSelector) (int, int, error) {
+	val, err := strconv.ParseInt(s.Vendor, 16, 32)
+	if err != nil {
+		return -1, -1, err
+	}
+	vendor := int(val)
+
+	val, err = strconv.ParseInt(s.Product, 16, 32)
+	if err != nil {
+		return -1, -1, err
+	}
+	product := int(val)
+
+	return vendor, product, nil
+}
+
+func DiscoverAllowedUSBDevices(usbs []v1.USBHostDevice) map[string][]*PluginDevices {
+	// The return value: USB USBDevice Plugins found and permitted to be exposed
+	plugins := make(map[string][]*PluginDevices)
+	// All USB devices found plugged in the Node
+	localDevices := discoverLocalUSBDevicesFunc()
+	for _, usbConfig := range usbs {
+		resourceName := usbConfig.ResourceName
+		// only accept ExternalResourceProvider: true for USB devices
+		if !usbConfig.ExternalResourceProvider {
+			logrus.Errorf("Skipping discovery of %s. To be handled by external device-plugin",
```

Comment:
in this case the device is being managed by the internal kubevirt device plugin, we should probably tweak the message to reflect the same 
```suggestion
			logrus.Errorf("Skipping discovery of %s. To be handled by kubevirt internally",
```